rootfs/etc/profile.d/_geodesic-dotfiles.sh

```diff
@@ -0,0 +1,84 @@
+# bash funnctions to import
+
+# Given the name of a resource, _search_geodesic_dirs assembles an array of matching, Geodesic-specific resources,
+# in order from most general to most specific. They should be applied in order with the later ones overriding
+# the earlier ones.
+#
+# Every resource, for example "preferences", can be either a file or a directory.
+# * If it is a file, it is loaded directly
+# * If it is a directory, all the non-hidden files in that directory are loaded in glob sort order
+#
+# Several directories are searched for resources, in this order:
+# * $base/defaults/ ($base itself defaults to /localhost/.geodesic, can be set via GEODESIC_DOT_DIR)
+# * $base/ if and only if there is no $base/defaults/ directory
+# * $base/$(dirname $DOCKER_IMAGE)/defaults/
+# * $base/$(dirname $DOCKER_IMAGE)/ if and only if there is no $base/$(dirname $DOCKER_IMAGE)/defaults/ directory
+# * $base/$(base $DOCKER_IMAGE)/
+# * $base/$DOCKER_IMAGE/
+#
+# Usage: _search_geodesic_dirs array-ref resource-name
+# The first argument to _search_geodesic_dirs is the name of a variable. That variable will be treated as an array
+# and all the files found will be added in the order described above. The second argument is the name of the resource
+# See example usages in _preferences.sh
+
+function _search_geodesic_dirs() {
+	local -n search_list=$1
+	local resource=$2
+	local base="${GEODESIC_DOT_DIR}"
```

Comment:
@aknysh @tamsky @goruha any thoughts on the name of this env?

- `GEODESIC_LOCAL_CONF`
- `GEODESIC_LOCAL_DIR`
- `GEODESIC_CONF_DIR` (probably confusing with `conf/`)
- `GEODESIC_DOT_DIR`
- `GEODESIC_DOT_CONF`
- `GEODESIC_......???`

Reply:
It's extra confusing already because it refers to a directory inside Geodesic ***and*** a directory on the Docker host computer.

Reply:
hence `GEODESIC_LOCAL_*` something might be less confusing?

Reply:
`GEODESIC_HOST_DIR` ?

Reply:
Prior to this PR, Geodesic stored bash history files in `~/.geodesic`, and would automatically add `~/.geodesic/env` as a Docker `--env-file` to `docker run`. There was no option to change this.  `GEODESIC_DOT_DIR` is just a new option so you can store these things in some directory other than `~/.geodesic`. It says to use this configured directory instead of the Geodesic dot dir. We could take this variable out altogether and it would not change anything. It's just a new option.